src/components/menu-item/MenuItem.tsx

```diff
@@ -1,50 +1,64 @@
 
-const MenuItem = ({
-    label,
-    description,
-    inlineDescription = false,
-    destructive = false,
-    disabled = false,
-    leadingElement,
-    trailingElementLabel,
-    trailingElement,
-}: PMenuItem): JSX.Element => {
-    const rootProperties = {
-        destructive,
-        disabled,
-    };
+const MenuItem = React.forwardRef(
+    (props: PMenuItem, reference: ForwardedRef<null>): JSX.Element => {
+        const {
+            label,
+            description,
+            inlineDescription = false,
+            destructive = false,
+            disabled = false,
+            leadingElement,
+            trailingElementLabel,
+            trailingElement,
+            onClick,
+            onHover,
+        } = props;
 
-    return (
-        <MenuItemRoot {...rootProperties}>
-            {leadingElement}
-            <Grid
-                flex={1}
-                row={inlineDescription}
-                alignItems={inlineDescription ? 'center' : 'initial'}
-                padding={Spacing.only('left', leadingElement ? 125 : 0)}
-            >
-                <MenuItemLabelRoot>{label}</MenuItemLabelRoot>
-                {description && (
-                    <MenuItemDescriptionRoot withMargin={!inlineDescription}>
-                        {description}
-                    </MenuItemDescriptionRoot>
-                )}
-            </Grid>
-            <Grid row alignItems={'center'} justifyItems={'flex-start'} alignSelf={'flex-start'}>
-                {trailingElement && trailingElementLabel && (
-                    <MenuItemDescriptionRoot>{trailingElementLabel}</MenuItemDescriptionRoot>
-                )}
-                {trailingElement}
-            </Grid>
-        </MenuItemRoot>
-    );
-};
+        const rootProperties = {
+            onMouseEnter: onHover,
+            onClick,
+            destructive,
+            disabled,
+        };
+
+        return (
+            <MenuItemRoot ref={reference} {...rootProperties}>
+                {leadingElement}
+                <Grid
+                    flex={1}
+                    row={inlineDescription}
+                    alignItems={inlineDescription ? 'center' : 'initial'}
+                    padding={Spacing.only('left', leadingElement ? 125 : 0)}
+                >
+                    <MenuItemLabelRoot>{label}</MenuItemLabelRoot>
+                    {description && (
+                        <MenuItemDescriptionRoot withMargin={!inlineDescription}>
+                            {description}
+                        </MenuItemDescriptionRoot>
+                    )}
+                </Grid>
+                <Grid
+                    row
+                    alignItems={'center'}
+                    justifyItems={'flex-start'}
+                    alignSelf={'flex-start'}
+                >
+                    {trailingElement && trailingElementLabel && (
```

Comment:
nit: this condition could be helpful as a assertion at the top to give the user a hint in the console what went wrong